zipkin-autoconfigure/collector-rabbitmq/src/test/java/zipkin/collector/rabbitmq/ZipkinRabbitMqCollectorPropertiesOverrideTest.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright 2015-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.collector.rabbitmq;
+
+import com.rabbitmq.client.Address;
+import java.util.Arrays;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.springframework.boot.autoconfigure.context.PropertyPlaceholderAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Configuration;
+import zipkin.autoconfigure.collector.rabbitmq.ZipkinRabbitMqCollectorProperties;
+import zipkin.collector.rabbitmq.RabbitMqCollector;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.boot.test.util.EnvironmentTestUtils.addEnvironment;
+
+@RunWith(Parameterized.class)
+public class ZipkinRabbitMqCollectorPropertiesOverrideTest {
+
+  AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+
+  @After
+  public void close() {
+    if (context != null) context.close();
+  }
+
+  @Parameterized.Parameter(0) public String property;
+  @Parameterized.Parameter(1) public Object value;
+  @Parameterized.Parameter(2) public Function<ZipkinRabbitMqCollectorProperties, Object>
+      propertiesExtractor;
+  @Parameterized.Parameter(3) public Function<RabbitMqCollector.Builder, Object> builderExtractor;
+
+  @Parameterized.Parameters(name = "{0}")
+  public static Iterable<Object[]> data() {
+    return Arrays.asList(
+        parameters("addresses", "localhost:5671, localhost:5673",
+          properties -> properties.getAddresses().toString().substring(1, properties.getAddresses().toString().length() - 1),
```

Comment:
This is a bit wonky due to the mismatch of types: the properties uses a `List<String>`, the builder `Address[]`, and of course we are passing just a String via the environment/YAML but Spring converts to a list for us. We could add a converter for `Address` but this felt like more work than it is worth right now.